pkg/nsx/services/firewall.go

```diff
@@ -535,22 +536,28 @@ func (service *SecurityPolicyService) buildRuleDstGroup(obj *v1alpha1.SecurityPo
 func (service *SecurityPolicyService) buildRuleServiceEntries(rulePorts *[]v1alpha1.SecurityPolicyPort) *[]*data.StructValue {
 	ruleServiceEntries := []*data.StructValue{}
 	for _, port := range *rulePorts {
+		var portRange string
+		startPort := port.Port.IntValue()
 		sourcePorts := data.NewListValue()
-		sourcePorts.Add(data.NewIntegerValue(int64(port.Port.IntValue())))
 		destinationPorts := data.NewListValue()
 		// In case that the destination_port in NSX-T is 0.
-		endPort := int64(port.EndPort)
+		endPort := port.EndPort
 		if endPort == 0 {
-			endPort = int64(port.Port.IntValue())
+			portRange = fmt.Sprint(startPort)
+		} else {
+			portRange = fmt.Sprintf("%d-%d", startPort, endPort)
 		}
-		destinationPorts.Add(data.NewIntegerValue(endPort))
+		destinationPorts.Add(data.NewStringValue(string(portRange)))
 		serviceEntry := data.NewStructValue(
 			"",
 			map[string]data.DataValue{
 				"source_ports":      sourcePorts,
 				"destination_ports": destinationPorts,
 				"l4_protocol":       data.NewStringValue(string(port.Protocol)),
 				"resource_type":     data.NewStringValue("L4PortSetServiceEntry"),
+				// adding the following default values to make it easy when compare the existing object from store and the new built object
+				"marked_for_delete": data.NewBooleanValue(false),
+				"overridden":        data.NewBooleanValue(false),
```

Comment:
Could we add a simplify struct for service entry like other resources?

Reply:
Yes, theoretically I think we can. However, if we simplify this property, we need to dive into `rule.ServiceEntries[i].source_ports/destination_ports/l4_protocol/resource_type`, i.e. about 3 layers of indexes. I thought doing it won't make other things can reuse it so I didn't do it. As I commented, this is a tricky method to "make it easy", do you think it's acceptable?

Reply:
ok. Later if we have more similar cases, maybe we find a better and consistent way.

Reply:
Thanks for understanding. If we want to solve it in a common way in the future, maybe need to consider make the change in `store.go`, i.e. reject some unused options when saving the object into the store.